PlaygroundService/Controllers/PlaygroundController.cs

```diff
@@ -86,29 +93,193 @@ public async Task<IActionResult> BuildService(IFormFile contractFiles)
+                using (var stream = new FileStream(tempFilePath, FileMode.Open))
+                {
+                    var fileId = await _gridFS.UploadFromStreamAsync(file.FileName, stream);
+                    System.IO.File.Delete(tempFilePath);
+                    Directory.Delete(extractionPath, true);
+                    return Ok(new { id = fileId.ToString() });
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "An error occurred while processing the uploaded ZIP file.");
+                return StatusCode(500, "An internal server error occurred.");
+            }
+            finally
+            {
+                System.IO.File.Delete(tempFilePath);
+                if (Directory.Exists(extractionPath))
+                {
+                    Directory.Delete(extractionPath, true);
+                }
+            }
+        }
+
+        private static List<string> ValidateFolderStructure(string extractionPath, Dictionary<string, List<string>> expectedStructure)
+        {
+            var missingFiles = new List<string>();
+
+            foreach (var folder in expectedStructure)
+            {
+                var folderPath = Path.Combine(extractionPath, folder.Key);
+                if (!Directory.Exists(folderPath))
+                {
+                    missingFiles.Add($"Folder '{folder.Key}' is missing.");
+                    continue; // Skip to the next folder since this one doesn't exist
+                }
+
+                foreach (var pattern in folder.Value)
+                {
+                    try
+                    {
+                        var matchedFiles = Directory.GetFiles(folderPath, pattern, SearchOption.AllDirectories);
+                        if (!matchedFiles.Any())
+                        {
+                            missingFiles.Add($"Pattern '{pattern}' is missing in '{folder.Key}' folder.");
+                        }
+                    }
+                    catch (DirectoryNotFoundException)
+                    {
+                        missingFiles.Add($"Directory not found for pattern '{pattern}' in folder '{folder.Key}'.");
+                    }
+                }
+            }
+
+            return missingFiles;
+        }
+
+        [HttpGet("share/get/{id}")]
+        public async Task<IActionResult> DownloadZipFile(string id)
+        {
+            if (!ObjectId.TryParse(id, out var objectId))
+            {
+                return BadRequest("Invalid ID.");
+            }
+
+            var stream = await _gridFS.OpenDownloadStreamAsync(objectId);
+
+            if (stream == null)
+            {
+                return NotFound("File not found.");
+            }
+
+            return File(stream, "application/zip", stream.FileInfo.Filename);
+        }
+
+        private static bool ScanFileForViruses(string filePath)
+        {
+            ProcessStartInfo psi = new ProcessStartInfo
```

Comment:
Use `RunProcess` in ProcessHelper instead and use it to replace all logic within the `ScanFileForViruses`. You can make the `RunProcess` method public.